capabilityreceiver/pkg/chplmapper/capabilityinfo.go

```diff
@@ -26,9 +26,55 @@ var fluffWords = []string{
 	"corporation.",
 }
 
+type details struct {
+	ID   int    `json:"id"`
+	Name string `json:"name"`
+}
+
+type chplEndpointListProductInfo struct {
+	ListSourceURL    string                 `json:"listSourceURL"`
+	SoftwareProducts []chplCertifiedProduct `json:"softwareProducts"`
+}
+
+type chplCertifiedProduct struct {
+	ChplProductNumber string  `json:"chplProductNumber"`
+	Developer         details `json:"developer"`
+}
+
+type chplMapResults struct {
+	ChplProductIDs []string
+	ChplDeveloper  string
+}
+
 // MatchEndpointToVendor creates the database association between the endpoint and the vendor,
 // and the endpoint and the healht IT product.
-func MatchEndpointToVendor(ctx context.Context, ep *endpointmanager.FHIREndpointInfo, store *postgresql.Store) error {
+func MatchEndpointToVendor(ctx context.Context, ep *endpointmanager.FHIREndpointInfo, store *postgresql.Store, chplProductInfoFile string) error {
+
+	listSourceMap, err := openCHPLEndpointListInfoFile(chplProductInfoFile)
```

Comment:
would it make sense to move this function to the capabilityreceiver? it's called both here and in the `MatchEndpointToProduct`, but I feel like you could call the function before you call `MatchEndpointToVendor` and `MatchEndpointtoProduct` in the `saveMsgInDb` function, and then you could just pass the map around to these functions instead of passing the file and having to recreate the map multiple times.